src/main/java/org/opentripplanner/routing/vehicle_parking/VehicleParking.java

```diff
@@ -0,0 +1,199 @@
+
+  /**
+   * Source specific tags of the vehicle parking, which describe the available features. For example
+   * park_and_ride, bike_lockers, or static_osm_data.
+   */
+  private final List<String> tags;
+
+  // TODO: this would need to be parsed from the OSM format
+  private final RepeatingTimePeriod openingHours;
+
+  private final RepeatingTimePeriod feeHours;
+
+  private final I18NString note;
+
+  @Builder.Default
+  private final VehicleParkingState state = VehicleParkingState.OPERATIONAL;
+
+  @Getter(AccessLevel.NONE)
+  private final boolean bicyclePlaces;
+
+  @Getter(AccessLevel.NONE)
+  private final boolean carPlaces;
+
+  @Getter(AccessLevel.NONE)
+  private final boolean wheelchairAccessibleCarPlaces;
+
+  @EqualsAndHashCode.Exclude
+  private VehiclePlaces capacity;
+
+  @EqualsAndHashCode.Exclude
+  private VehiclePlaces availability;
+
+  @Builder.Default
+  private final List<VehicleParkingEntrance> entrances = new ArrayList<>();
+
+  public String toString() {
+    return String.format(Locale.US, "VehicleParking(%s at %.6f, %.6f)", name, y, x);
+  }
+
+  /**
+   * The number of spaces by type. {@code null} if unknown.
+   */
+  @Data
+  @Builder
+  public static class VehiclePlaces implements Serializable {
+
+    private final Integer bicycleSpaces;
+
+    private final Integer carSpaces;
+
+    private final Integer wheelchairAccessibleCarSpaces;
+  }
+
+  /**
+   * The state of the vehicle parking. TEMPORARILY_CLOSED and CLOSED are distinct states so that
+   * they may be represented differently to the user.
+   */
+  public enum VehicleParkingState {
+    OPERATIONAL,
+    TEMPORARILY_CLOSED,
+    CLOSED
+  }
+
+  public boolean hasBicyclePlaces() {
+    return bicyclePlaces;
+  }
+
+  public boolean hasAnyCarPlaces() {
+    return hasCarPlaces() || hasWheelchairAccessibleCarPlaces();
+  }
+
+  public boolean hasCarPlaces() {
+    return carPlaces;
+  }
+
+  public boolean hasWheelchairAccessibleCarPlaces() {
+    return wheelchairAccessibleCarPlaces;
+  }
+
+  public boolean hasRealTimeData() {
+    return availability != null;
+  }
+
+  public void updateVehiclePlaces(VehiclePlaces vehiclePlaces) {
+    this.availability = vehiclePlaces;
+    this.capacity = vehiclePlaces;
+  }
+
+  private void addEntrance(VehicleParkingEntranceCreator creator) {
+    var entrance = creator.updateValues(VehicleParkingEntrance.builder()
+            .vehicleParking(this))
+            .build();
+
+    entrances.add(entrance);
+  }
+
+  @Getter
+  @Builder
+  @EqualsAndHashCode
+  public static class VehicleParkingEntrance implements Serializable {
+
+    @EqualsAndHashCode.Exclude
+    private final VehicleParking vehicleParking;
+
+    private final FeedScopedId entranceId;
+
+    private final double x, y;
+
+    private final I18NString name;
+
+    // Used to explicitly specify the intersection to link to instead of using (x, y)
+    @EqualsAndHashCode.Exclude
+    private final transient StreetVertex vertex;
```

Comment:
I merged this PR and strangely I'm observing the graph serialization test to fail: https://github.com/mfdz/OpenTripPlanner/runs/2682942115?check_suite_focus=true#step:6:3086

What is the reason to make this field transient?

Reply:
Because when I remove the transient keyword, the test passes.

Reply:
I've added a commit to explicitly clear the value of `VehicleParkingEntrance#vertex`, which is used when linking entrances to OSM.

The test fails due to #3910 modifying the builtin Portland graph to serialize bike rental stations and park and ride places. `ObjectDiffer` doesn't ignore transient fields, and so the missing transient value is caught after loading the graph and comparing it to the version in-memory.